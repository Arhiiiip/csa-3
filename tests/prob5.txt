(setq n 2520) (setq i 20) (setq x 0) (while (> i 0) (if (= (% n i) 0) (setq x 1) (setq i (- i 1))) (if (= x 1) (setq x 0) (setq i 20)) (if (= i 20) (setq x 0) (setq n (+ n 2520))) (if (= i 1) (setq x 0) (print n)))