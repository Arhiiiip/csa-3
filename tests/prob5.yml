source: |-
  (setq n 2520) (setq i 20) (setq x 0) (while (> i 0) (if (= (% n i) 0) (setq i (- i 1)) (setq x 0))(if (!= (% n i) 0) (setq n (+ n 2520)) (setq x 0))(if (!= (% n i) 0) (setq i 20) (setq x 0)) (if (= i 1) (print n) (setq x 0)))

output: |
  Output buffer: 555 | ticks: 555 | amount_instr: 555