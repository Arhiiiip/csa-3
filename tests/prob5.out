[
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx3",
        "arg2": 2520
    },
    {
        "opcode": "wr",
        "arg1": "rx3"
    },
    {
        "opcode": "ld",
        "arg1": "rx4",
        "arg2": 0
    },
    {
        "opcode": "wr",
        "arg1": "rx4"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx5",
        "arg2": 20
    },
    {
        "opcode": "wr",
        "arg1": "rx5"
    },
    {
        "opcode": "ld",
        "arg1": "rx6",
        "arg2": 0
    },
    {
        "opcode": "wr",
        "arg1": "rx6"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 4
    },
    {
        "opcode": "ld",
        "arg1": "rx7",
        "arg2": 0
    },
    {
        "opcode": "wr",
        "arg1": "rx7"
    },
    {
        "opcode": "ld",
        "arg1": "rx8",
        "arg2": 0
    },
    {
        "opcode": "wr",
        "arg1": "rx8"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 67
    },
    {
        "opcode": "ld",
        "arg1": "rx9",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx10",
        "arg2": "rx2"
    },
    {
        "opcode": "jle",
        "arg1": "rx10",
        "arg2": "rx9"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 35
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx11",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx3",
        "arg2": "rx2"
    },
    {
        "opcode": "div",
        "arg1": "rx3",
        "arg2": "rx11"
    },
    {
        "opcode": "ld",
        "arg1": "rx4",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 33
    },
    {
        "opcode": "jne",
        "arg1": "rx14",
        "arg2": "rx4"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 4
    },
    {
        "opcode": "ld",
        "arg1": "rx5",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 35
    },
    {
        "opcode": "jmp"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx5",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 50
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx6",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx7",
        "arg2": "rx2"
    },
    {
        "opcode": "div",
        "arg1": "rx7",
        "arg2": "rx6"
    },
    {
        "opcode": "ld",
        "arg1": "rx8",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 48
    },
    {
        "opcode": "je",
        "arg1": "rx14",
        "arg2": "rx8"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 4
    },
    {
        "opcode": "ld",
        "arg1": "rx9",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 50
    },
    {
        "opcode": "jmp"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx9",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 65
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx10",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx11",
        "arg2": "rx2"
    },
    {
        "opcode": "div",
        "arg1": "rx11",
        "arg2": "rx10"
    },
    {
        "opcode": "ld",
        "arg1": "rx3",
        "arg2": 0
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 63
    },
    {
        "opcode": "je",
        "arg1": "rx14",
        "arg2": "rx3"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 4
    },
    {
        "opcode": "ld",
        "arg1": "rx4",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 65
    },
    {
        "opcode": "jmp"
    },
    {
        "opcode": "ld",
        "arg1": "rx2",
        "arg2": 2
    },
    {
        "opcode": "ld",
        "arg1": "rx4",
        "arg2": "rx2"
    },
    {
        "opcode": "ld",
        "arg1": "rx15",
        "arg2": 15
    },
    {
        "opcode": "jmp"
    },
    {
        "opcode": "halt"
    }
]